spec/models/access_spec.rb

```diff
@@ -0,0 +1,636 @@
+require "rails_helper"
+
+RSpec.describe Access, type: :model do
```

Comment:
There's a ton of tests in here (6 times more LOC than the code itself) and they may seem a bit redundant, but they aren't, they are mostly every small case documented explicitly.

Reply:
Rigorous testing of this important piece of code is pretty expected, I think :)